app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,49 @@
+class AuthController < ApplicationController
+  # Required for OmniAuth Dev Flow... Not to be used in Production
+  skip_before_action :verify_authenticity_token, only: :callback unless Rails.env.production?
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    
+    if session[:user_id]
+      # Means our user is signed in. Add the authorization to the user
+      User.find(session[:user_id]).add_provider(auth_hash)
+ 
+      render :text => "You can now login using #{auth_hash["provider"].capitalize} too!"
+    else
+      # Log him in or sign him up
+      auth = Authorization.find_or_create(auth_hash)
+ 
+      # Create the session
+      session[:user_id] = auth.user.id
+   
+      render :text => "Welcome #{auth.user.name}!"
+    end
+    redirect_to session[:redirect_path] || '/'
```

Comment:
We should use route helpers here rather than '/' - I believe the helper you want is root_path.